ethereum/p2p/src/test/java/org/hyperledger/besu/ethereum/p2p/rlpx/wire/WireMessagesSedesTest.java

```diff
@@ -59,6 +67,8 @@ private static void assertSedesWorks(final byte[] data) {
     // Re-serialize and check that data matches
     final BytesValueRLPOutput out = new BytesValueRLPOutput();
     peerInfo.writeTo(out);
-    assertThat(out.encoded()).isEqualTo(input);
+    if (encEqualsInput) {
```

Comment:
Should probably only re-serialise the data if we're going to actually check it matches.

Reply:
I've created an issue to look at encoding and decoding of 0. This will probably be reversed when this ticket has been finished.